src/cmd/services/m3aggregator/serve/serve.go

```diff
@@ -39,22 +43,38 @@ func Serve(
 	doneCh chan struct{},
 	iOpts instrument.Options,
 ) error {
-	log := rawTCPServerOpts.InstrumentOptions().Logger().Sugar()
+	log := iOpts.Logger()
 	defer aggregator.Close()
 
-	rawTCPServer := rawtcpserver.NewServer(rawTCPAddr, aggregator, rawTCPServerOpts)
-	if err := rawTCPServer.ListenAndServe(); err != nil {
-		return fmt.Errorf("could not start raw TCP server at %s: %v", rawTCPAddr, err)
+	if m3msgAddr != "" {
+		m3msgServer, err := m3msgserver.NewServer(m3msgAddr, m3msgServerOpts, aggregator, iOpts)
+		if err != nil {
+			return fmt.Errorf("could not create m3msg server: %v", err)
```

Comment:
nit: maybe include Addr in the log here? Alternatively, would it be a good idea to have `log = log.With(zap.String("m3msgAddress", m3msgAddr))` and similar for other server types?

Reply:
Sounds good, agreed